src/main/java/com/conveyal/datatools/manager/jobs/AutoDeployJob.java

```diff
@@ -0,0 +1,188 @@
+                    continue;
+                }
+
+                if (deployment.latest() == null) {
+                    String message = String.format("Deployment `%s` has never been deployed. Skipping auto-deploy for project `%s`.",
+                        deployment.name,
+                        project.name);
+                    LOG.warn(message);
+                    NotifyUsersForSubscriptionJob.createNotification(
+                        "project-updated",
+                        project.id,
+                        message
+                    );
+                    continue;
+                }
+                List<String> feedVersionIdsForDeployment = getFeedVersionsForDeployment(project);
+                if (feedVersionIdsForDeployment.isEmpty()) {
+                    LOG.info("Project {} skipped for auto deployment as no newer feed versions are available!",
+                        project.name);
+                    continue;
+                }
+                // Send deployment (with new feed version) to most recently used server.
+                String latestServerId = deployment.latest().serverId;
+                OtpServer server = Persistence.servers.getById(latestServerId);
+                if (server != null) {
+                    deployment.feedVersionIds = new ArrayList<>();
+                    deployment.feedVersionIds.addAll(feedVersionIdsForDeployment);
+                    // If there are no other fetches in progress, queue up the deploy job.
+                    if (DeploymentController.queueDeployJob(new DeployJob(deployment, owner, server))) {
+                        LOG.debug("Last auto deploy date updated for project {}.", project.name);
+                        project.lastAutoDeploy = new Date();
+                        Persistence.projects.replace(project.id, project);
+                    } else {
+                        LOG.debug("Could not auto-deploy to {} due to conflicting active deployment for project {}.",
+                            server.name,
+                            project.name);
+                    }
+                } else {
+                    LOG.debug("Server with id {} no longer exists. Skipping deployment for project {}.",
+                        latestServerId,
+                        project.name);
+                }
+            } finally {
+                lockedProjects.remove(project.id);
+                LOG.debug("Auto deploy lock removed for project id: {}", project.id);
+            }
+        }
+        status.completeSuccessfully("Auto deploy complete.");
+    }
+
+    /**
+     * Retrieve feed versions from all feed sources which have no critical errors and where created after the last
+     * auto deploy.
+     */
+    private List<String> getFeedVersionsForDeployment(Project project) {
+        List<String> feedVersionIds = new ArrayList<>();
+        Collection<FeedSource> feedSources = project.retrieveProjectFeedSources();
```

Comment:
Retrieving all of the feed sources from the project is going to ignore and override the current deployment configuration's feed sources. Only feed versions within feed sources that have been added to a deployment configuration should be used.

Reply:
I've removed this method. I'm not 100% on this, but providing `canAutoDeployProject` comes back true - the deployment can go ahead.

Reply:
Have addressed in the latest commit.

Reply:
Ended up being addressed in Evans PR. Closing.